test/Test/Database/LSMTree/Internal/RawOverflowPage.hs

```diff
@@ -0,0 +1,58 @@
+module Test.Database.LSMTree.Internal.RawOverflowPage (
+    -- * Main test tree
+    tests,
+) where
+
+import qualified Data.Primitive.ByteArray as BA
+import qualified Data.Vector.Primitive as PV
+
+import           Test.QuickCheck
+import           Test.QuickCheck.Instances ()
+import           Test.Tasty (TestTree, testGroup)
+import           Test.Tasty.QuickCheck
+
+import           Database.LSMTree.Generators (LargeRawBytes (..))
+import           Database.LSMTree.Internal.RawOverflowPage
+import           Database.LSMTree.Internal.Serialise.RawBytes (RawBytes (..))
+import qualified Database.LSMTree.Internal.Serialise.RawBytes as RawBytes
+
+tests :: TestTree
+tests =
+  testGroup "Database.LSMTree.Internal.RawOverflowPage"
+    [ testProperty "RawBytes prefix to RawOverflowPage"
+                   prop_rawBytesToRawOverflowPage
+    , testProperty "RawBytes to [RawOverflowPage]"
+                   prop_rawBytesToRawOverflowPages
+    ]
+
+-- | Converting up to the first 4096 bytes of a 'RawBytes' to an
+-- 'RawOverflowPage' and back gives us the original, padded with zeros to a
+-- multiple of the page size.
+prop_rawBytesToRawOverflowPage :: LargeRawBytes -> Property
+prop_rawBytesToRawOverflowPage
+  (LargeRawBytes bytes@(RawBytes (PV.Vector off len ba))) =
+    label (if RawBytes.size bytes >= 4096 then "large" else "small") $
+    label (if BA.isByteArrayPinned ba then "pinned" else "unpinned") $
+    label (if off == 0 then "offset 0" else "offset non-0") $
+
+        rawOverflowPageRawBytes (makeRawOverflowPage ba off len)
+    === RawBytes.take 4096 bytes <> padding
+  where
+    padding = RawBytes.fromVector (PV.replicate paddinglen 0)
+    paddinglen
+      | RawBytes.size bytes > 4096 = 0
+      | otherwise = 4096 - (RawBytes.size bytes `mod` 4096)
```

Comment:
Isn't this just `4096 - RawBytes.size bytes`? You could add a `max 0`, but `replicate` should handle the negative value fine.
Actually, for a size of 4096, I think your code gives 4096 bytes of padding?